spec/datadog/core/environment/execution_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+require 'datadog/core/environment/execution'
+
+RSpec.describe Datadog::Core::Environment::Execution do
+  describe '#development?' do
+    subject(:development?) { described_class.development? }
+
+    context 'when in an RSpec test' do
+      it { is_expected.to eq(true) }
+    end
+
+    context 'when not in an RSpec test' do
+      # RSpec is detected through the $PROGRAM_NAME.
+      # Changing it will make RSpec detection to return false.
+      #
+      # We change the $PROGRAM_NAME instead of stubbing
+      # `Datadog::Core::Environment::Execution.rspec?` because
+      # otherwise we'll have no real test for non-RSpec cases.
+      around do |example|
+        begin
+          original = $PROGRAM_NAME
+          $PROGRAM_NAME = 'not-rspec'
+          example.run
+        ensure
+          $PROGRAM_NAME = original
+        end
+      end
```

Comment:
This made me think that one possible alternative implementation strategy is to maybe have a proc-per-thing we want to detect, e.g.

```ruby
RSPEC_PROBE = proc { $PROGRAM_NAME.end_with?(RSPEC_PROGRAM_NAME) }
MINITEST_PROBE = proc { ... }

def test?(test_probes: [RSPEC_PROBE, MINITEST_PROBE])
  test_probes.any?(&:call)
end
```

then we could override individual probes (or even dependency inject them).

This to say -- it came to mind; I think the current version is great as well so as usual with my random suggestions, feel free to ignore :)

Reply:
I agree that this suggested approach looks good!

I think the file is small now to keep it as it, but it's definitely a great alternative.